frontend/src/ui/Access.tsx

```diff
@@ -670,6 +762,112 @@ const ActionMenuItem: React.FC<ActionMenuItemProps> = (
+    const user = useUser();
+    const resetModalRef = useRef<ModalHandle>(null);
+
+    const containsUser = (acl: Acl) =>
+        isRealUser(user) && (userIsOwner || user.roles.some(r =>
+            r === COMMON_ROLES.ADMIN
+            || acl.get(r)?.actions.has("write")
+            || acl.get(r)?.actions.has("admin")
+            || inheritedAcl?.get(r)?.actions.has("admin"))
+        );
+
+    const selectionIsInitial = selections.size === initialAcl.size
+        && [...selections].every(([role, info]) => {
+            const other = initialAcl.get(role);
+            return other && areSetsEqual(other.actions, info.actions);
+        });
+
+    const submit = onSubmit;
+
+    useNavBlocker(!selectionIsInitial);
+
+    return (
+        <div {...{ className }} css={{
+            display: "flex",
+            gap: 8,
+            alignSelf: "flex-start",
+            marginTop: 40,
+        }}>
+            {/* Reset button */}
+            <Button
+                disabled={selectionIsInitial}
+                onClick={() => currentRef(resetModalRef).open()}
+                css={{ ...!selectionIsInitial && { ":hover": { color: COLORS.danger0 } } }}
+            >
+                {t("manage.access.reset-modal.label")}
+            </Button>
+            <Modal ref={resetModalRef} title={t("manage.access.reset-modal.title")}>
+                <p>{t("manage.access.reset-modal.body")}</p>
+                <div css={{
+                    display: "flex",
+                    gap: 12,
+                    justifyContent: "center",
+                    flexWrap: "wrap",
+                    marginTop: 32,
+                }}>
+                    <Button onClick={() => currentRef(resetModalRef).close?.()}>
+                        {t("general.action.cancel")}
+                    </Button>
+                    <Button kind="danger" onClick={() => {
+                        setSelections(initialAcl);
+                        currentRef(resetModalRef).close?.();
+                    }}>{t("manage.access.reset-modal.label")}</Button>
+                </div>
+            </Modal>
+
+            {/* Save button */}
+            <Button
+                disabled={selectionIsInitial}
+                onClick={() =>
+                    !containsUser(selections)
+                        ? currentRef(saveModalRef).open()
+                        : submit(selections)
+                }
+                css={{ ...!selectionIsInitial && { ":hover": { color: COLORS.happy0 } } }}
+            >{t("general.action.save")}</Button>
+            {inFlight && <div css={{ marginTop: 16 }}><Spinner size={20} /></div>}
+            <ConfirmationModal
+                ref={saveModalRef}
+                title={t("manage.access.save-modal.title")}
+                buttonContent={t("manage.access.save-modal.confirm")}
+                onSubmit={() => submit(selections)}
+            >
+                <p>{t("manage.access.save-modal.body")}</p>
```

Comment:
This unconditionally mentions "write-access". At this point I'm fine with that. The intentions are clear. Ideally it probably should say "page admin privilege" in the realm cases... but again, I don't think it's too important, just wanted to remark it here.